contracts/src/CrossChainApplications/ERC20Bridge/ERC20Bridge.sol

```diff
@@ -456,19 +465,17 @@ contract ERC20Bridge is IERC20Bridge, ITeleporterReceiver, ReentrancyGuard {
      * @dev Teleporter message receiver for minting of an existing bridge token on this chain.
      *
      * Emits a {MintBridgeTokens} event.
+     *
+     * Note: This function is only called within `receiveTeleporterMessage`, which can only be
+     * called by the Teleporter messenger.
      */
     function _mintBridgeTokens(
         bytes32 nativeChainID,
         address nativeBridgeAddress,
         address nativeContractAddress,
         address recipient,
         uint256 amount
-    ) private nonReentrant {
-        // Only allow the Teleporter messenger to deliver messages.
-        if (msg.sender != address(teleporterMessenger)) {
-            revert Unauthorized();
-        }
-
+    ) private nonReentrant onlyAllowedTeleporter {
```

Comment:
Is the `onlyAllowedTeleporter` modifier needed here? I think it ends up being redundant given that `_mintBridgeTokens` can only be called by `receiveTeleporterMessage` which already has the same modifier (as noted in your comment). Same with `_transferBridgeTokens` below.

I think it could potentially cause unnecessary confusion around where it is needed and where it is not.

Reply:
I was questioning the same thing, originally added it because thought to make it clear that those functions should only be called by teleporter, but think if it's made clear that these functions are only called by `receiveTelepoterMessage` we can remove for redundancy. Removed.